src/EditorFeatures/Test2/InlineParameterNameHints/AbstractInlineParameterNameHintsTests.vb

```diff
@@ -0,0 +1,54 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.CSharp.InlineParameterNameHints
+Imports Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
+Imports Microsoft.CodeAnalysis.Editor.Shared.Extensions
+Imports Microsoft.CodeAnalysis.Editor.Shared.Utilities
+Imports Microsoft.CodeAnalysis.Editor.Tagging
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.InlineParameterNameHints
+Imports Microsoft.CodeAnalysis.Shared.TestHooks
+Imports Microsoft.VisualStudio.Text
+Imports Microsoft.VisualStudio.Text.Tagging
+
+Namespace Microsoft.CodeAnalysis.Editor.UnitTests.InlineParameterNameHints
+    <[UseExportProvider]>
+    Public MustInherit Class AbstractInlineParameterNameHintsTests
+
+        Protected Async Function VerifyParamHints(test As XElement, Optional optionIsEnabled As Boolean = True) As Tasks.Task
+            Using workspace = TestWorkspace.Create(test)
+                WpfTestRunner.RequireWpfFact($"{NameOf(AbstractInlineParameterNameHintsTests)}.{NameOf(Me.VerifyParamHints)} creates asynchronous taggers")
+
+                Dim hostDocument = workspace.Documents.Single()
```

Comment:
We're only grabbing a single document here, but then we're also looping through all documents later down; what's the intent here?